osm2lanes-web/src/control.rs

```diff
@@ -1,36 +1,84 @@
 use std::cell::RefCell;
+use std::collections::BTreeMap;
 use std::rc::Rc;
 
-use osm2lanes::locale::{Country, DrivingSide};
+use gloo_timers::callback::Timeout;
+use osm2lanes::locale::{Country, DrivingSide, Locale};
+use osm2lanes::test::{get_tests, TestCase};
 use web_sys::{Event, FocusEvent, HtmlInputElement, HtmlSelectElement, KeyboardEvent, MouseEvent};
 use yew::{html, Callback, Component, Context, Html, NodeRef, Properties, TargetCast};
 
-use crate::{Msg, State};
+use crate::{Msg as AppMsg, State};
+
+pub enum Msg {
+    Up(Box<AppMsg>),
+    FirstLazy,
+    Example(String),
```

Comment:
Just checking my understanding here... `Up` refers to the parent component, `FirstLazy` is a trick to lazily load up test examples, and `Example` is when a particular example has been set

Reply:
yeah, naming is hard